numpy/core/src/multiarray/lowlevel_strided_loops.c.src

```diff
@@ -995,34 +1006,52 @@ PyArray_GetStridedNumericCastFn(int aligned, npy_intp src_stride,
 #  if NPY_USE_UNALIGNED_ACCESS
                     if (src_stride == sizeof(@type1@) &&
                                 dst_stride == sizeof(@type2@)) {
-                        return &_aligned_contig_cast_@name1@_to_@name2@;
+                        return &_aligned_contig_cast_@name1@_to_@name2@@isa@;
                     }
                     else {
-                        return &_aligned_cast_@name1@_to_@name2@;
+                        return &_aligned_cast_@name1@_to_@name2@@isa@;
```

Comment:
Meant to do a review, but then posted instead:

IIRC this branch is never used (`NPY_USE_UNALIGNED_ACCESS` is always 0 here) and I don't think vectorization is OK if it was used.  So I would either not do this, or just delete the whole `# if` block:  It doesn't really add a whole lot.

Reply:
I am unsure of the meaning of NPY_USE_UNALIGNED_ACCESS, but if we can remove it here since it is always set to 0, why not removing it from the whole file and possibly the whole code ? It would make the code a bit more clean/readable.

Reply:
I have to look closer, but I think we should just delete this code (if you agree with that).